required/latex-lab/latex-lab-footnotes.dtx

```diff
@@ -0,0 +1,1731 @@
+%    in the same place, etc.
+%
+%
+% \subsubsection{Configuration points}
+%
+%    None: everything is implemented through a single definition for
+%    \cs{@footnotemark} that offers a number of hooks that can be used
+%    by packages to implement handling of multiple marks and the
+%    formatting of marks.
+%
+%
+% \subsubsection{Hooks}
+%
+%    The hooks to customize the marks in the text are the following:
+%    \begin{description}
+%    \item[\hook{fnmark/before}]
+%
+%      Executed at the very beginning of \cs{footnotemark}. Currently
+%      there are two packages (\pkg{bibarts} and \pkg{chextras}) that
+%      prepend material at this point (not ncessarily correctly, e.g.,
+%      they do not all check that they are in horizontal mode).
+%
+%      This hook is paired with hook \hook{fnmark/after}.
+%
+%    \item[\hook{fnmark}]
+%
+%      Executed in horizontal mode and after the current space factor
+%      has been saved away for reuse. This is where currently code for multiple
+%      marks does its preparation (as done by \pkg{footmisc} and
+%      others).
+%
+%      The hook is only executed in hmode, i.e., not if the mark is
+%      generated in math --- maybe that means the multiple handling
+%      should happen later?
+%
+%      After the hook \cs{nobreak} is executed, so any
+%      ``material'' added in the hook is tied to the following mark
+%      unless it contains its own permissible penalty.
+%
+%    \item[\hook{fnmark/begin}]
+%
+%      This hook is executed directly in front of the typeset mark.
+%      This is the place where \pkg{hyperref} adds part of its code,
+%      i.e., after the \cs{nobreak} mentioned above. With a closer
+%      integration of \pkg{hyperref} this hook may not be necessary at
+%      all.
+%
+%    \item[\hook{fnmark/end}]
+%
+%      This hook is executed directly after the typeset mark. It is
+%      used by \pkg{hyperref}, \pkg{memhfixc}, \pkg{scrlttr2}, and
+%      \pkg{footmisc}. Used, for example, to implement support for
+%      multiple marks in succession.
+%
+%      It is \emph{not} a reversed hook.
+%
+%
+%    \item[\hook{fnmark/after}]
+%
+%      This hook is executed at the very end of the \cs{footnotemark} command.
+%
+%      It is a reversed  hook to pair with \hook{fnmark/before}
+%    \end{description}
+%
+%
+%
+%
+% \subsubsection{Additional configuration possibilities}
+%
+%    The actual formatting is done through \cs{@makefnmark} --- no
+%    special customization support for now.
+%
+%
+%
+% \subsection{Formatting the footnote text}
+%
+%    This implements the formatting of the footnote text the way it
+%    appears at the bottom of the page (default case), or possibly
+%    elsewhere, e.g. in the margin.
+%
+% \subsubsection{Configuration points}
+%
+%    To cater for different layout configurations there are four
+%    configuration points which can be set only by one package or
+%    class. These are:
```

Comment:
What you say is correct, there should be a full-blown documentation on config points (but there isn't yet). And there should be more examples, I agree. However, we should keep in mind that in contrast to hooks, there are only very very few packages ever, that touch configuration points. Those here, for example would only be altered by footmisc (or its equivalent if it goes into the format) and "perhaps" KOMA and memoir if they intend to supply their own footnote handling for paragraph footnotes and that's it. Other packages and classes all just supply code for one of the hooks (or rather right now they overwrite `\makefntext`unconditionally instead of doing so) but not attempt to alter LaTeX's overall footnote handling. Even `manyfooot` or `bigfoot`really tried to do any of that but relied on footmisc to do that part.

So I have been a bit economical with the documentation for now, given priorities. Anybody, who want to see how they are used in detail can study the footmisc reimplementation using them.